rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/transform/TbRenameMsgKeysNodeTest.java

```diff
@@ -0,0 +1,161 @@
+    TbRenameKeysNode node;
+    TbRenameKeysNodeConfiguration config;
+    TbNodeConfiguration nodeConfiguration;
+    TbContext ctx;
+    TbMsgCallback callback;
+
+    @BeforeEach
+    void setUp() throws TbNodeException {
+        deviceId = new DeviceId(UUID.randomUUID());
+        callback = mock(TbMsgCallback.class);
+        ctx = mock(TbContext.class);
+        config = new TbRenameKeysNodeConfiguration().defaultConfiguration();
+        config.setRenameKeysMapping(Map.of("TestKey_1", "Attribute_1", "TestKey_2", "Attribute_2"));
+        nodeConfiguration = new TbNodeConfiguration(JacksonUtil.valueToTree(config));
+        node = spy(new TbRenameKeysNode());
+        node.init(ctx, nodeConfiguration);
+    }
+
+    @AfterEach
+    void tearDown() {
+        node.destroy();
+    }
+
+    @Test
+    void givenDefaultConfig_whenInit_thenOK() {
+        assertThat(node.config).isEqualTo(config);
+    }
+
+    @Test
+    void givenDefaultConfig_whenVerify_thenOK() {
+        TbRenameKeysNodeConfiguration defaultConfig = new TbRenameKeysNodeConfiguration().defaultConfiguration();
+        assertThat(defaultConfig.getRenameKeysMapping()).isEqualTo(Collections.emptyMap());
+        assertThat(defaultConfig.isFromMetadata()).isEqualTo(false);
+    }
+
+    @Test
+    void givenMsg_whenOnMsg_thenVerifyOutput() throws Exception {
+        String data = "{\"Temperature_1\":22.5,\"TestKey_2\":10.3}";
+        node.onMsg(ctx, getTbMsg(deviceId, data));
+
+        ArgumentCaptor<TbMsg> newMsgCaptor = ArgumentCaptor.forClass(TbMsg.class);
+        verify(ctx, times(1)).tellSuccess(newMsgCaptor.capture());
+        verify(ctx, never()).tellFailure(any(), any());
+
+        TbMsg newMsg = newMsgCaptor.getValue();
+        assertThat(newMsg).isNotNull();
+
+        JsonNode dataNode = JacksonUtil.toJsonNode(newMsg.getData());
+        assertThat(dataNode.has("Attribute_2")).isEqualTo(true);
+        assertThat(dataNode.has("Temperature_1")).isEqualTo(true);
+    }
+
+    @Test
+    void givenMetadata_whenOnMsg_thenVerifyOutput() throws Exception {
+        config = new TbRenameKeysNodeConfiguration().defaultConfiguration();
+        config.setRenameKeysMapping(Map.of("TestKey_1", "Attribute_1", "TestKey_2", "Attribute_2"));
+        config.setFromMetadata(true);
+        nodeConfiguration = new TbNodeConfiguration(JacksonUtil.valueToTree(config));
+        node.init(ctx, nodeConfiguration);
+
+        String data = "{\"Temperature_1\":22.5,\"TestKey_2\":10.3}";
+        node.onMsg(ctx, getTbMsg(deviceId, data));
+
+        ArgumentCaptor<TbMsg> newMsgCaptor = ArgumentCaptor.forClass(TbMsg.class);
+        verify(ctx, times(1)).tellSuccess(newMsgCaptor.capture());
+        verify(ctx, never()).tellFailure(any(), any());
+
+        TbMsg newMsg = newMsgCaptor.getValue();
+        assertThat(newMsg).isNotNull();
+
+        Map<String, String> mdDataMap = newMsg.getMetaData().getData();
+        assertThat(mdDataMap.containsKey("Attribute_1")).isEqualTo(true);
```

Comment:
verify that test validation is correct

Reply:
if I'm understand correctly there should be: Attribute_2

Reply:
This test checks the copying from the metadata `config.setFromMetadata(true);`, and the key "TempKey_1" in the metadata